tests/forte/data/data_store_test.py

```diff
@@ -307,6 +307,46 @@ def test_get_type_info(self):
         DataStore._type_attributes = self.reference_type_attributes
         # TODO: need more tests for ontology file input
 
+    def test_entry_methods(self):
+        sent_type = "ft.onto.base_ontology.Sentence"
+        doc_type = "ft.onto.base_ontology.Document"
+        ann_type = "forte.data.ontology.top.Annotation"
+        group_type = "forte.data.ontology.top.Group"
+        sent_list = list(self.data_store._DataStore__elements[sent_type])
+        doc_list = list(self.data_store._DataStore__elements[doc_type])
+        ann_list = (
+            doc_list
+            + sent_list
+            + list(self.data_store._DataStore__elements[ann_type])
+        )
+        group_list = list(self.data_store._DataStore__elements[group_type])
+        sent_entries = list(self.data_store.all_entries(sent_type))
+        doc_entries = list(self.data_store.all_entries(doc_type))
+        ann_entries = list(self.data_store.all_entries(ann_type))
+
+        self.assertEqual(sent_list, sent_entries)
+        self.assertEqual(doc_list, doc_entries)
+        self.assertEqual(ann_list, ann_entries)
+
+        num_sent_entries = self.data_store.num_entries(sent_type)
+        num_doc_entry = self.data_store.num_entries(doc_type)
+        num_ann_entry = self.data_store.num_entries(ann_type)
+
+        self.assertEqual(num_sent_entries, len(sent_list))
+        self.assertEqual(num_doc_entry, len(doc_list))
+        self.assertEqual(num_ann_entry, len(ann_entries))
+
+        # remove a sentence
+        self.data_store.delete_entry(9999)
+        num_sent_entries = self.data_store.num_entries(sent_type)
+
+        self.assertEqual(num_sent_entries, len(sent_list) - 1)
+
+        # remove a group
+        self.data_store.delete_entry(23456)
+        num_group_entries = self.data_store.num_entries(group_type)
+        self.assertEqual(num_group_entries, len(group_list) - 1)
+
```

Comment:
add a few more tests:
1. delete more than one entry
2. delete an entry of a child type and count the number of the parent type
3. try delete and then add, and then count the numbers